etna/metrics/intervals_metrics.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+
+from etna.datasets import TSDataset
+from etna.metrics.base import Metric
+from etna.metrics.base import MetricAggregationMode
+
+
+def dummy():
+    return np.nan
+
+
+class Coverage(Metric):
+    """Coverage metric for prediction intervals."""
+
+    def __init__(
+        self, quantiles: Tuple[float, float] = (0.025, 0.975), mode: str = MetricAggregationMode.per_segment, **kwargs
+    ):
+        """
+        Precenteage of samples in the interval [lower quantile, upper quantile].
+
+        .. math::
+            Coverage(y\_true, y\_pred) = \\frac{\\sum_{i=0}^{n-1}{[ y\_true_i \\ge y\_pred_i^{lower\_quantile}] * [y\_true_i \\le y\_pred_i^{upper\_quantile}] }}{n}
+
+
+        Notes
+        -----
+        Works just if quantiles presented in y_pred
+
+        Parameters
+        ----------
+        mode: 'macro' or 'per-segment'
+            metrics aggregation mode
+        kwargs:
+            metric's computation arguments
+        """
+        super().__init__(mode=mode, metric_fn=dummy, **kwargs)
+        self.quantiles = quantiles
+
+    def __call__(self, y_true: TSDataset, y_pred: TSDataset) -> Union[float, Dict[str, float]]:
+        """
+        Compute metric's value with y_true and y_pred.
+
+        Notes
+        -----
+        Note that if y_true and y_pred are not sorted Metric will sort it anyway
+
+        Parameters
+        ----------
+        y_true:
+            dataset with true time series values
+        y_pred:
+            dataset with predicted time series values
+
+        Returns
+        -------
+            metric's value aggregated over segments or not (depends on mode)
+        """
+        self._validate_segment_columns(y_true=y_true, y_pred=y_pred)
+
+        segments = set(y_true.df.columns.get_level_values("segment"))
+        metrics_per_segment = {}
+        for segment in segments:
+            self._validate_timestamp_columns(
+                timestamp_true=y_true[:, segment, "target"].dropna().index,
+                timestamp_pred=y_pred[:, segment, "target"].dropna().index,
+            )
+            upper_quantile_flag = y_true[:, segment, "target"] <= y_pred[:, segment, f"target_{self.quantiles[1]:.4g}"]
```

Comment:
looks like we need to check that `f"target_{self.quantiles[1]:.4g}"` is in y_pred

Reply:
why should we format quantiles values like `f"target_{self.quantiles[1]:.4g}"`?

Reply:
> looks like we need to check that `f"target_{self.quantiles[1]:.4g}"` is in y_pred

I've added validation for it and test case

Reply:
> why should we format quantiles values like `f"target_{self.quantiles[1]:.4g}"`?

I don't remember exactly where issues could arise. But in some cases due to float point arithmetic we had bad quantile representation, so we've started using such formatting